File: src/Semester.java
import java.util.Random;

public class Semester {
    private String semesterName;
    private int semesterNo;

    public String getSemesterName() {
        return defineSemesterName(getSemesterNo());
    }

    public int getSemesterNo() {
        return semesterNoGenerate();
    }
    public boolean equals(Semester obj) {
        return semesterNo == obj.semesterNo;
    }

    private void setSemesterName() {
        return;
    }

    public int compareTo(Semester obj) {
        return 1;
    }

    public String toString() {
        return "";
    }
    public int semesterNoGenerate(){
        Random random=new Random();
        return random.nextInt(8-1)+1;
    }
    public String defineSemesterName(int semesterNo){
        if(semesterNo%2 == 0)
            semesterName = "SPRING";
        else
            semesterName = "FALL";
        return semesterName;

    }
}

File: iteration-3/Semester.py
class Semester:

    def __init__(self, semesterNo:str, semesterName:str):
        self._semesterNo = semesterNo
        self._semesterName = semesterName


    @property
    def semesterNo(self):
        return self._semesterNo
    
    @semesterNo.setter
    def semesterNo(self, semesterNo):
        self._semesterNo = semesterNo

    @property
    def semesterName(self):
        return self._semesterName
    
    @semesterName.setter
    def semesterName(self, semesterName):
        self._semesterName = semesterName